modules/gdscript/editor/script_templates/CharacterBody2D/basic_movement.gd

```diff
@@ -19,9 +19,6 @@ func _physics_process(delta: float) -> void:
 	# Get the input direction and handle the movement/deceleration.
 	# As good practice, you should replace UI actions with custom gameplay actions.
 	var direction := Input.get_axis("ui_left", "ui_right")
-	if direction:
-		velocity.x = direction * SPEED
-	else:
-		velocity.x = move_toward(velocity.x, 0, SPEED)
+	velocity.x = direction * SPEED
```

Comment:
Why this change? It ignores the deceleration, this significantly changes the behaviour, which isn't a "clean up"

Reply:
It makes no difference to the behavior; can you check if the commit description helps you understand the issue?

Reply:
No it absolutely changes it, I've used that code, how did you establish it doesn't?

Reply:
>  move_toward calls are equivalent to
assigning the velocity components to 0

This is false, *unless* the correct fix is to do `SPEED * delta` for the method

Reply:
`velocity` can never be greater than speed because `direction` is normalized, which means the code is equivalent to:
```
if direction:
    velocity.x = direction * SPEED
else:
    velocity.x = 0
```

Which itself is equivalent to `velocity.x = direction * SPEED`

Reply:
Then the correct code is `move_toward(velocity.x, 0, SPEED * delta)`

Reply:
I updated the description with a clearer wording

Reply:
Then this implies a lot more changes because with the current `SPEED` the friction is way too low, and we should probably split that constant `SPEED` into a deceleration constant.

I thought this kind of improvement to the code could be added on top of this PR which is meant to clarify the current state of the code (admittedly loosing the original intention of the code, which I couldn't get looking at it)

Reply:
> I thought this kind of improvement

It's not an improvement if the code is broken, and if it is then the bug should be fixed, if anything should be done, otherwise I don't think there's anything in this PR which significantly improves the state of this, and should be done in a PR fixing the code instead

Reply:
Otherwise I'll just go ahead and fix this tomorrow, if you don't feel confident doing so

Reply:
Commit replaced with the introduction of a deceleration constant